starter-analytics-postgres/src/test/groovy/io/micronaut/starter/analytics/postgres/StoreGeneratedProjectStatsSpec.groovy

```diff
@@ -48,10 +52,50 @@ class StoreGeneratedProjectStatsSpec extends Specification implements TestProper
          "datasources.default.dialect": Dialect.POSTGRES.name()]
     }
 
+    @Inject UnauthorizedAnalyticsClient unauthorizedClient
+    @Inject WrongApiKeyClient wrongApiKeyClient
     @Inject AnalyticsClient client
     @Inject ApplicationRepository repository
     @Inject FeatureRepository featureRepository
 
+    void "test save generation data without api key"() {
+        given:
+        def generated = new Generated(
+                ApplicationType.FUNCTION,
+                Language.KOTLIN,
+                BuildTool.MAVEN,
+                TestFramework.SPOCK,
+                MicronautJdkVersionConfiguration.DEFAULT_OPTION
+        )
+        generated.setSelectedFeatures([new SelectedFeature("google-cloud-function")])
+
+        when:
+        unauthorizedClient.applicationGenerated(generated).get()
+
+        then:
+        Exception e = thrown()
```

Comment:
use `HttpClientResponseException` which I think is the exception being thrown and check .getStatus() == 401